### PR TITLE
avcodec/jpeg2000: Fix FF_DWT97_INT to pass the conformance testing defined in ISO/IEC 15444-4

### PR DESCRIPTION
**Pull request by osamu620**
avcodec/jpeg2000: Fix FF_DWT97_INT to pass the conformance testing defined in ISO/IEC 15444-4

Fix for the integer version of the inverse 9-7 DWT processing (FF_DWT97_INT, https://trac.ffmpeg.org/ticket/10123), which is activated with `-flags +bitexact`.

I went through the code path for the DWT 9-7 transform (integer) and improved precision to match conformance codestream.

As a result, the encoded codestream size is slightly larger for a given Q value. For example, `-flags +bitexact -i lena.pnm -q: 20 -format j2k -y tmp.j2c` gives 13K (HEAD) and 19K (with this patch).

This commit also updates the source and reference files for affected FATE tests.

### CODE REVIEW

File: libavcodec/jpeg2000.c [hunk 1]
```diff
@@ -268,7 +268,7 @@ static void init_band_stepsize(AVCodecContext *avctx,
         av_log(avctx, AV_LOG_ERROR, "stepsize out of range\n");
     }
 
-    band->i_stepsize = lrint(band->f_stepsize * (1 << 15) + 0.5f);
+    band->i_stepsize = (int)floorf(band->f_stepsize * (1 << 15));
```
Comment (osamu620) [inline libavcodec/jpeg2000.c:271]: @palemieux With this correction, I got the following results.

```
❯ ./dwt97-diff.sh ~/Documents/data/images/ppm/ElephantDream_small.ppm 0
.rw-rw-r-- 1,497,613 osamu 21 Oct 16:39 encoded-HEAD.j2c
.rw-rw-r-- 1,498,160 osamu 21 Oct 16:39 encoded-PATCH.j2c
PSNR decoded by exact-HEAD
90.2224 (0.902224)
PSNR decoded by exact-PATCH
91.6198 (0.916198)
PSNR decoded by float-HEAD
79.0372 (0.790372)
PSNR decoded by float-PATCH
81.9933 (0.819933)
```

Now, the mismatch between float97 and int97 seems to have disappeared. For both float and int, the larger codestream size gives better quality.

Comment (palemieux) [reply libavcodec/jpeg2000.c:271]: Looks great. Two questions:

* why move `#define F_LFTG_ALPHA` to a difference source file?
* why use `//` sometimes and `/*` some other times?

**Review by palemieux — CHANGES_REQUESTED** [review]

File: libavcodec/jpeg2000.c (hunk 1 above)
Comment (osamu620) [reply libavcodec/jpeg2000.c:271]: Thanks for reviewing!

- A1: I thought `jpeg2000dwt.h` was included in `tests/jpeg2000dwt.c`. But I found that `jpeg2000dwt.c` was included. So, this change makes no sense. I would revert this in the next commit.

- A2: No reason exists. In the next commit, I will unify the commenting style into `//`.

Comment (osamu620) [reply libavcodec/jpeg2000.c:271]: @palemieux 
> Looks great. Two questions:
> 
> * why move `#define F_LFTG_ALPHA` to a difference source file?

`I_PRESHIFT` should remain in `jpeg2000dwt.h`. Do you agree with this?

> * why use `//` sometimes and `/*` some other times?

The mixture of `//` and `/* */` has existed for a long time. Should we unify all?

Comment (palemieux) [reply libavcodec/jpeg2000.c:271]: > I_PRESHIFT should remain in jpeg2000dwt.h. Do you agree with this?

Yes.

Comment (palemieux) [reply libavcodec/jpeg2000.c:271]: >The mixture of // and /* */ has existed for a long time. Should we unify all?

Not on this patch. The patch should be consistent with itself or with the code surrounding it.

**Review by palemieux — APPROVED** [review]

**Pull request closed without merging**